ams/optimizer/flink-optimizer/src/main/java/com/netease/arctic/optimizer/flink/FlinkExecutor.java

```diff
@@ -51,8 +51,13 @@ public void open() throws Exception {
         getRuntimeContext().getTaskManagerRuntimeInfo().getConfiguration().getString(HA_CLUSTER_ID);
     executor = (FlinkOptimizerExecutor) allExecutors[subTaskIndex];
     // set optimizer flink runtime info, including application_id, tm_id, host
-    executor.addRuntimeContext("application_id", applicationId);
-    executor.addRuntimeContext("tm_id", taskManagerId);
+    if (applicationId != null) {
+      executor.addRuntimeContext("application_id", applicationId);
+    }
+    if (taskManagerId != null) {
+      executor.addRuntimeContext("tm_id", taskManagerId);
+    }
+    executor.addRuntimeContext("subtask_index", String.valueOf(subTaskIndex));
```

Comment:
`subtask_index` doesn't seem to help determine which taskmanager it is, I can't see it on the flink web or in the logs.

Reply:
> `subtask_index` doesn't seem to help determine which taskmanager it is, I can't see it on the flink web or in the logs.

thank you for your reply.
![image](https://github.com/apache/incubator-amoro/assets/40817998/1532779c-06e1-42ac-8946-eb3d43d2298e)
You can judge from here

Reply:
Get.